homeassistant/components/palazzetti/climate.py

```diff
@@ -0,0 +1,159 @@
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PalazzettiConfigEntry
+from .const import (
+    ACTION_NOT_UNAVAILABLE,
+    DOMAIN,
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_MODES,
+    FAN_SILENT,
+    PALAZZETTI,
+)
+from .coordinator import PalazzettiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: PalazzettiConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Palazzetti climates based on a config entry."""
+    async_add_entities(
+        [
+            PalazzettiClimateEntity(
+                coordinator=entry.runtime_data,
+            )
+        ]
+    )
+
+
+class PalazzettiClimateEntity(
+    CoordinatorEntity[PalazzettiDataUpdateCoordinator], ClimateEntity
+):
+    """Defines a Palazzetti climate."""
+
+    _attr_has_entity_name = True
+    _attr_hvac_modes = []  # The available modes will be set when we know the current state
+    _attr_fan_modes = FAN_MODES
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_translation_key = DOMAIN
+
+    def __init__(self, *, coordinator: PalazzettiDataUpdateCoordinator) -> None:
+        """Initialize Palazzetti climate."""
+        super().__init__(coordinator=coordinator)
+        client = coordinator.client
+        self._attr_unique_id = f"{coordinator.config_entry.data[CONF_MAC]}-climate"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.config_entry.data[CONF_MAC])},
+            name=coordinator.config_entry.data[CONF_NAME],
+            manufacturer=PALAZZETTI,
+            sw_version=client.sw_version,
+            hw_version=client.hw_version,
+        )
+        self._attr_name = coordinator.config_entry.data[CONF_NAME]
+        self._attr_fan_modes = list(
+            map(str, range(client.fan_speed_min, client.fan_speed_max + 1))
+        )
+        if self.coordinator.client.has_fan_silent:
+            self._attr_fan_modes.insert(0, FAN_SILENT)
+        if self.coordinator.client.has_fan_high:
+            self._attr_fan_modes.append(FAN_HIGH)
+        if self.coordinator.client.has_fan_auto:
+            self._attr_fan_modes.append(FAN_AUTO)
+
+    @property
+    def available(self) -> bool:
+        """Is the entity available."""
+        return self.coordinator.client.connected
+
+    @property
+    def supported_features(self) -> ClimateEntityFeature:
+        """Supported features."""
+        features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+        )
+        if self.coordinator.client.has_on_off_switch:
```

Comment:
Is this calculated dynamicly?

Reply:
I'm removing this, to adopt @zweckj 's recommandation to not make this dynamic anymore.